openml-h2o/src/main/java/com/feedzai/openml/h2o/H2OModelCreator.java

```diff
@@ -133,13 +143,42 @@ public AbstractClassificationH2OModel loadModel(final Path modelPath,
             );
         }
 
+        validateSchema(genModel, schema);
         final AbstractClassificationH2OModel resultingModel = createModel(modelPath, schema, genModel, closeable);
         ClassificationValidationUtils.validateClassificationModel(schema, resultingModel);
 
         logger.info("Model loaded successfully.");
         return resultingModel;
     }
 
+    /**
+     * Performs a validation from the H2O model against the schema provided.
+     *
+     * <p>
+     *     Note that the field names stored in the {@link IGeneratedModel provided model} do not include the target variable (i.e. response field/column).
+     *     As such, if the {@link DatasetSchema schema provided} contains a target variable, it is not taken into consideration upon this validation
+     *     and correspondent error message (it case the schemas are not compatible.
+     * </p>
+     *
+     * @param genModel H2O model to validate.
+     * @param schema   The schema which the model will be bound to.
+     * @throws ModelLoadingException thrown if the schema from the model is not compatible with the one provided.
+     */
+    private void validateSchema(final IGeneratedModel genModel, final DatasetSchema schema) throws ModelLoadingException {
+        final Set<String> expectedFields = Sets.newHashSet(genModel.getNames());
+        final Optional<FieldSchema> targetOpt = schema.getTargetFieldSchema();
+        final Set<String> schemaFields = schema.getFieldSchemas().stream()
+                .filter(field -> !targetOpt.isPresent() || Objects.equals(targetOpt.get().getFieldName(), field.getFieldName()))
+                .map(FieldSchema::getFieldName)
+                .collect(Collectors.toSet());
+
+        if (!Objects.equals(expectedFields, schemaFields)) {
+            final String errMsg = String.format("The model contains fields '%s' (size %d), but the schema contains '%s' (size %d).", expectedFields, expectedFields.size(),
+                    schemaFields, schemaFields.size());
+            throw new ModelLoadingException(errMsg);
```

Comment:
Please log before throwing; I like that practice since I never trust upstream code (it might swallow the exception :) )